### PR TITLE
[Inference] Add validated models for Gaudi

### PR DESCRIPTION
**Pull request by Deegue**
### Model list:
bloom-7b1 single card without template
Falcon-7b single card without template
Falcon-40b multiple cards without template
Gemma-2b single card without template
Llama3-7b single card unknow
Llama3-70b multiple cards unknow
Mistral-7b single card without template
Mixtral-8x7B-Instruct-v0.1 single card with template
llama-2-7b single card unknow
llama-2-70b multiple cards unknow
CodeLlama single card unknow
GPT2 single card without template
GPT-J single card without template
MPT-7b single card without template
Qwen1.5-110B single card with template

### CODE REVIEW

File: llm_on_ray/inference/models/hpu/MindChat-Qwen2-4B-hpu.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+port: 8000
+name: MindChat-Qwen2-4B 
+route_prefix: /MindChat-Qwen2-4B 
+num_replicas: 1
+cpus_per_worker: 8
+hpus_per_worker: 1
+device: hpu
+model_description:
+  model_id_or_path: X-D-Lab/MindChat-Qwen2-4B
```
Comment (carsonwang) [inline llm_on_ray/inference/models/hpu/MindChat-Qwen2-4B-hpu.yaml:9]: Not this one. Please use Qwen/Qwen1.5-7B

File: llm_on_ray/inference/models/hpu/CodeLlama-7b-hf-hpu.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+name: CodeLlama-7b-hf
+route_prefix: /CodeLlama-7b-hf
+num_replicas: 1
+cpus_per_worker: 8
+hpus_per_worker: 1
+device: hpu
+model_description:
+  model_id_or_path: codellama/CodeLlama-7b-hf
+  tokenizer_name_or_path: codellama/CodeLlama-7b-hf
+  chat_processor: ChatModelGptJ
```
Comment (carsonwang) [inline llm_on_ray/inference/models/hpu/CodeLlama-7b-hf-hpu.yaml:11]: This config file is not correct. Please refer to the existing yaml file for CPU.

**Review by carsonwang — COMMENTED** [review]
@Deegue, although the result showed pass but they actually failed. Please check the run result. So can you please check the response of the model for the tests.

**Comment by Deegue** [comment]
All CI passed. Gentle ping @carsonwang for review, thanks!

**Comment by carsonwang** [comment]
@kira-lin is helping to review this. For qwen, can you please update to use Qwen/Qwen2-7B-Instruct?

**Comment by Deegue** [comment]
> @kira-lin is helping to review this. For qwen, can you please update to use Qwen/Qwen2-7B-Instruct?

Added `Qwen1.5-7B-Chat` and `Qwen2-7B-Instruct`.

**Review by kira-lin — CHANGES_REQUESTED** [review]
Qwen 110 not tested.
Many models output weird things, including mpt mistral gpt2 gemma falcon 7b/40b bloom.
Compare them with CPU/GPU to see if it's normal behavior. For example, codellama is outputting some markup language, but I think it's tuned that way. Also pay attention to temperature, which might lead to random result, and it might be different for different models

**Comment by kira-lin** [comment]
For falcon and qwen, @KepingYan can look into this.

File: llm_on_ray/inference/models/hpu/Qwen1.5-110B-hpu.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+port: 8000
```
Comment (carsonwang) [inline llm_on_ray/inference/models/hpu/Qwen1.5-110B-hpu.yaml:1]: Please remove this file.

File: llm_on_ray/inference/models/hpu/Qwen1.5-7B-Chat-hpu.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+port: 8000
```
Comment (carsonwang) [inline llm_on_ray/inference/models/hpu/Qwen1.5-7B-Chat-hpu.yaml:1]: Please also remove this file.

File: llm_on_ray/inference/models/hpu/Mixtral-7B-hpu.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+port: 8000
```
Comment (carsonwang) [inline llm_on_ray/inference/models/hpu/Mixtral-7B-hpu.yaml:1]: Also remove this one. Add it back after running successfully.

**Review by KepingYan — CHANGES_REQUESTED** [review]
https://github.com/intel/llm-on-ray/blob/cdce225cb2285baf9c151fecc7b6af853412e030/llm_on_ray/inference/predictors/hpu_predictor.py#L340

Let's modify this line according to:
https://github.com/huggingface/optimum-habana/blob/595cc3e4ec219b1ce469b323cf94e994c5c5d8f3/examples/text-generation/utils.py#L311-L312

**Comment by Deegue** [comment]
> https://github.com/intel/llm-on-ray/blob/cdce225cb2285baf9c151fecc7b6af853412e030/llm_on_ray/inference/predictors/hpu_predictor.py#L340
> 
> Let's modify this line according to: https://github.com/huggingface/optimum-habana/blob/595cc3e4ec219b1ce469b323cf94e994c5c5d8f3/examples/text-generation/utils.py#L311-L312

Updated, thanks for comment. Btw, will it still be any place to be changed since I found some other places specially handled through `model_type == llama`?

**Review by carsonwang — APPROVED** [review]

**Pull request closed without merging**